### ISSUE TITLE
How to reproduce the result in the paper.

### ISSUE THREAD

**Issue by yuchenlichuck** [dvlab-research/Stratified-Transformer#12]
I run for the model from scratch and the best mIoU is: 0.701

**Comment by X-Lai** [comment]
Have you tested the model using the command illustrated in the README.md file?

**Comment by X-Lai** [comment]
Please be reminded that the validation process in `train.py` is not the same as 'test.py', since the validation in `train.py` does not reproject the predictions back to the original points.

**Comment by yuchenlichuck** [comment]
Okay, I will try my best models with the _test.py_

**Comment by yuchenlichuck** [comment]
I have trained some experiments with A100 GPU. Here is my result on val:

<google-sheets-html-origin><style type="text/css"></style>

66.78095238 | 66.9 | 66.2 | 68.2 | 66 | 66.7 | 67.3 | 66.7 | 66.2 | 65.2 | 66.7 | 66.3 | 66.6 | 67.5 | 66.7 | 67.8 | 66.1 | 65.3 | 67.9 | 67.9 | 67.7 | 66.2
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --


And I choose one best models for test.py and get the results:
```
39528:[2022-04-26 04:15:52,751 INFO test.py line 328 47644] Val result: mIoU/mAcc/allAcc 0.6953/0.7543/0.9064.
39529:[2022-04-26 04:15:52,751 INFO test.py line 329 47644] Val1 result: mIoU/mAcc/allAcc 0.6953/0.7543/0.9064.
```

**Comment by yuchenlichuck** [comment]
How to reopen the issue, I haven't seen the reopen button here.

**Comment by X-Lai** [comment]
Have you correctly used the configuration file and the command? It's wired to get a testing result that is even lower than the validation result. The 70.1 miou validation result seems reasonable, but your testing result is incorrect.

**Comment by X-Lai** [comment]
Here is my experiment log.
```

[2021-10-09 20:51:20,247 INFO test.py line 444 46792] Val result: mIoU/mAcc/allAcc 0.7196/0.7809/0.9147.
[2021-10-09 20:51:20,247 INFO test.py line 445 46792] Val1 result: mIoU/mAcc/allAcc 0.7196/0.7809/0.9147.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_0 Result: iou/accuracy 0.9623/0.9768, name: ceiling.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_1 Result: iou/accuracy 0.9871/0.9925, name: floor.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_2 Result: iou/accuracy 0.8562/0.9424, name: wall.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_3 Result: iou/accuracy 0.0000/0.0000, name: beam.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_4 Result: iou/accuracy 0.4613/0.5612, name: column.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_5 Result: iou/accuracy 0.5998/0.6109, name: window.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_6 Result: iou/accuracy 0.7675/0.8960, name: door.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_7 Result: iou/accuracy 0.8446/0.9192, name: chair.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_8 Result: iou/accuracy 0.9259/0.9695, name: table.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_9 Result: iou/accuracy 0.7518/0.7896, name: bookcase.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_10 Result: iou/accuracy 0.7779/0.8883, name: sofa.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_11 Result: iou/accuracy 0.7810/0.8258, name: board.
[2021-10-09 20:51:20,248 INFO test.py line 448 46792] Class_12 Result: iou/accuracy 0.6398/0.7793, name: clutter.
[2021-10-09 20:51:20,248 INFO test.py line 449 46792] <<<<<<<<<<<<<<<<< End Evaluation <<<<<<<<<<<<<<<<<
```

**Comment by yuchenlichuck** [comment]
I haven't modified the config except that I chose to use a single A100 GPU for each experiment. Another difference may be because the s3dis dataset preprocessing method that I used is different from yours. I need to double-check and run the experiments, thanks for your intermediate reply

**Comment by yuchenlichuck** [comment]
I used your pre-trained models, and test, and here is the results:

'''
2022-04-30 04:17:15,327 INFO test.py line 299 39911] Test: [68/68]-685639 Batch 156.143 (578.089) Accuracy 0.9243.
[2022-04-30 04:17:18,633 INFO test.py line 328 39911] Val result: mIoU/mAcc/allAcc 0.7276/0.7862/0.9191.
[2022-04-30 04:17:18,633 INFO test.py line 329 39911] Val1 result: mIoU/mAcc/allAcc 0.7276/0.7862/0.9191.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_0 Result: iou/accuracy 0.9630/0.9769, name: ceiling.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_1 Result: iou/accuracy 0.9877/0.9927, name: floor.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_2 Result: iou/accuracy 0.8661/0.9494, name: wall.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_3 Result: iou/accuracy 0.0000/0.0000, name: beam.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_4 Result: iou/accuracy 0.4851/0.5668, name: column.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_5 Result: iou/accuracy 0.6127/0.6234, name: window.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_6 Result: iou/accuracy 0.7820/0.9075, name: door.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_7 Result: iou/accuracy 0.8485/0.9221, name: chair.
[2022-04-30 04:17:18,633 INFO test.py line 332 39911] Class_8 Result: iou/accuracy 0.9274/0.9712, name: table.
[2022-04-30 04:17:18,634 INFO test.py line 332 39911] Class_9 Result: iou/accuracy 0.7607/0.8026, name: bookcase.
[2022-04-30 04:17:18,634 INFO test.py line 332 39911] Class_10 Result: iou/accuracy 0.7842/0.8897, name: sofa.
[2022-04-30 04:17:18,634 INFO test.py line 332 39911] Class_11 Result: iou/accuracy 0.7922/0.8317, name: board.
[2022-04-30 04:17:18,634 INFO test.py line 332 39911] Class_12 Result: iou/accuracy 0.6495/0.7872, name: clutter.
'''

**Comment by X-Lai** [comment]
This is because current `test.py` uses multiple test-time augmentations, which would lead to higher performance. The reported miou in our paper doesn't use any test-time augmentation.

**Comment by yuchenlichuck** [comment]
Okay, I use val miou=0.701 code can achieve 0.716 in test

**Comment by X-Lai** [comment]
The results are reasonable because S3DIS is relatively small and this incurs large variation.

**Comment by LuZaiJiaoXiaL** [comment]
Instead of setting the loop to 30, I trained the model with loop =1 to decrease training time. After around 100 epochs, the validation loss also did not decrease, but I found the mIoU_val was just around 44% rather than 70%. So does the loop of data really matter? Setting the loop to 30 may take seven days to finish the training. The efficiency improvement counts too much on time cost if so.
By the way, the experiment was implemented on a GPU 3090 (24G).
